### PR TITLE
Add audio files to pyfar

### LINKED ISSUE #304: Audio signals for quick listening?
Do we want to add audio signals for quick listening? I thought of something like a speech and music signals and a few HRTFs. Would be for convenience mostly but would not take much disk space.

### PR DESCRIPTION
**Pull request by f-brinkmann**
### Which issue(s) are closed by this pull request?

Closes #304 

### Changes proposed in this pull request:

add some useful audio signals to pyfar

Note: The drum signal is not mastered yet. But this will only change a gain factor in the code.

### CODE REVIEW

**Comment by sikersten** [comment]
Good basis to start with adding audio files!

- I am struggling a little with these functions to be part of the `signals` module. In contrast to the functions implemented so far, which generate signals, loading audio files is conceptually different.
- We could have these functions in a submodule of `signals` (similar as for `dsp`) or as separate module. So far, module names are without "_". Ideas for alternatives to `audio_files`: `files`, `examples`, `database`...
- Could you think of a reason to have a second HRIR database in the future? If yes, I'd suggest to be more distinct functions names, e.g., `fabian_hrir`.
- Add an example on how to do "quick listening"?

**Comment by f-brinkmann** [comment]
- I can live with both, having the audio files in `pf.signals` (because they are also signals) or in a different module (because they are read from disk rather than generated). Maybe discuss this during the weekly?
- I would prefer to only have one HRIR dataset in pyfar but I don't have a strong preference for the FABIAN set. I used it because I know it well and because it has the full horizontal and median plane as a subset. In any case, we could link to sofaconventions.org from the docstring in case someone wants to use a different set than we provide in pyfar.
- A *quick listening* example is a good idea. I would provide that either in the concepts or on top of the module description.

**Comment by f-brinkmann** [comment]
Will be moved to `pyfar.signals.files`. Will include:

- [ ] HpTF (FABIAN Database, HD-600 or similar)
- [ ] Loudspeaker IR (BRAS, Genelec 8020c)
- [ ] RIR (BRAS, Konzerthaus Berlin)
- [ ] BRIR (BRAS, Konzerthaus Berlin)
- [ ] Speech (Male, Female)
- [ ] Guitar (TH Köln)
- [ ] Castantes (M. Frank)
- [ ] Drums (..)

File: pyfar/signals/files.py [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+
+        **License**: CC BY-NC-SA 4.0, David Ackermann, Audio Communication
+        Group, Technical University of Berlin
+
+    Parameters
+    ----------
+    diffuse_field_compensation : bool, optional
+        Apply a diffuse field compensation to the BRIR. This can be used as a
+        simple headphone compensation filter when listening to the BRIR. The
+        default is False, which does not apply the compensation. The diffuse
```
Comment (sikersten) [inline pyfar/signals/files.py:231]:
```suggestion
        default is ``False``, which does not apply the compensation. The diffuse
```

File: pyfar/signals/files.py [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+        field compensation is taken from :py:func:`~hrirs`
```
Comment (sikersten) [inline pyfar/signals/files.py:232]: I think it's not really clear, that the compensation filter is part of the FABIAN database (I would have expected the inversion as part of the function here)

**Review by sikersten — COMMENTED** [review]
Looks good, I think it is a nice feature, thanks for taking care of it.
Only have minor comments on the docstrings, otherwise approved

File: pyfar/signals/files.py [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+    # resample brir
+    if sampling_rate != 48000:
+        guitar = pf.dsp.resample(guitar, sampling_rate)
+
+    return guitar
+
+
+def speech(voice="female", sampling_rate=44100):
+    """
+    Get anechoic speech sample.
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply pyfar/signals/files.py]: yes

File: pyfar/signals/files.py (hunk 2 above)
Comment (f-brinkmann) [reply pyfar/signals/files.py]: We would need an entire spherical HRTF data set to calculate in on the fly. I thus used the pre-calculated.

File: pyfar/signals/files.py (hunk 1 above)
Comment (f-brinkmann) [reply pyfar/signals/files.py]: yes.

File: pyfar/signals/files.py [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+        for resampling to the desired sampling rate.
+
+    Returns
+    -------
+    brir : Signal
+        The BRIR
+
+    References
+    ----------
+    .. [#] http://dx.doi.org/10.14279/depositonce-15774
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply pyfar/signals/files.py:245]: Yes, it's reserved and will be online as soon as the data is published.

File: pyfar/signals/files.py [hunk 5]
```diff
@@ -0,0 +1,515 @@
...
+    position : list, str, optional
+        The positions for which HRIRs are returned. HRIRs are available on the
+        horizontal and median plane in an angular resolution of 2 degrees.
+
+        ``'horizontal'``
+            Return horizontal plane HRIRs with an angular resolution of 2
+            degrees.
+        ``'median'``
+            Return median plane HRIRs with an angular resolution of 2 degrees.
+        List of coordinates
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply pyfar/signals/files.py:343]: What would you expect from full - all HRTFs that are part of pyfar or a full-spherical set? The latter would take more space, we currently only have the median/horizontal plane subset.

File: pyfar/signals/files.py [hunk 6]
```diff
@@ -0,0 +1,515 @@
...
+            Return HRIRs at specific positions defined by a list of azimuth
+            and elevation values in degrees. For example
+            ``[[30, 0], [330, 0]]`` returns HRIRs on the horizontal plane
+            (0 degree elevation) for azimuth angles of 30 and 330 degrees.
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply pyfar/signals/files.py:347]: yes, added that below.

File: pyfar/signals/files.py [hunk 7]
```diff
@@ -0,0 +1,515 @@
...
+        hrirs = pf.dsp.resample(hrirs, sampling_rate, 'freq')
+
+    return hrirs, sources
+
+
+def rir(sampling_rate=48000):
+    """
+    Get a room impulse response (RIR).
+
+    The RRIR was recorded with class I 1/2 inch measurement microphpone in the
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply pyfar/signals/files.py]: ups...

**Review by f-brinkmann — COMMENTED** [review]
@sikersten: all changes comitted

File: pyfar/signals/audio_files/hrirs.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
+"""
+Get selected data from FABIAN HRTF database and copy to pyfar
+http://dx.doi.org/10.14279/depositonce-5718.5
+"""
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply pyfar/signals/audio_files/hrirs.py]: Updated the file in its original location. Deleting it locally and loading it again should fix that.

File: pyfar/signals/files.py [hunk 9]
```diff
@@ -0,0 +1,516 @@
+"""
+This module contains anechoic audio content and impulse responses for listening
+and illustration. Note that each file has a separate license that is listed
+below.
+"""
```
Comment (sikersten) [inline pyfar/signals/files.py:5]: What about a short example on quick listening?
```suggestion

Quick listening is, e.g., possible with `sounddevice
<https://python-sounddevice.readthedocs.io/en/0.4.4/index.html>`_ installed:

>>> import pyfar as pf
>>> import sounddevice as sd
>>> # Load, illustrate and play speech signal
>>> speech = pf.signals.files.speech()
>>> pf.plot.spectrogram(speech)
>>> sd.play(speech.time.T, speech.sampling_rate)
"""
```

Comment (f-brinkmann) [reply pyfar/signals/files.py:15]: absolutely - added that

**Review by sikersten — APPROVED** [review]

File: pyfar/signals/files.py [hunk 10]
```diff
@@ -0,0 +1,526 @@
...
+    speech.time *= gain
+
+    # resample brir
+    if sampling_rate != 44100:
+        speech = pf.dsp.resample(speech, sampling_rate, post_filter=True)
+
+    return speech
+
+
+def brir(diffuse_field_compensation=False, sampling_rate=48000):
```
Comment (mberz) [inline pyfar/signals/files.py:222]: I'd vote in favor of not using abbreviations as function names. The dark ages of DOS restrictions on string lengths are over :)
```suggestion
def binaural_room_impulse_response(diffuse_field_compensation=False, sampling_rate=48000):
```

File: pyfar/signals/files.py [hunk 11]
```diff
@@ -0,0 +1,526 @@
...
+        brir = pf.dsp.convolve(brir, inverse_ctf, 'cut')
+
+    # resample brir
+    if sampling_rate != 48000:
+        brir = pf.dsp.resample(brir, sampling_rate, post_filter=True)
+
+    return brir
+
+
+def hpirs(sampling_rate=44100):
```
Comment (mberz) [inline pyfar/signals/files.py:286]: See comment above

File: pyfar/signals/files.py [hunk 12]
```diff
@@ -0,0 +1,526 @@
...
+    # load HRIRs
+    hpirs, *_ = pf.io.read_sofa(os.path.join(file_dir, files[0]))
+
+    if sampling_rate != 44100:
+        hpirs = pf.dsp.resample(hpirs, sampling_rate, 'freq', post_filter=True)
+
+    return hpirs
+
+
+def hrirs(position=[[0, 0]], diffuse_field_compensation=False,
```
Comment (mberz) [inline pyfar/signals/files.py:328]: See comment above

File: pyfar/signals/files.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (mberz) [reply pyfar/signals/files.py:353]: Let's not use the full data-set. Maybe as a compromise we can include a link where users can download the full data-set?

File: pyfar/signals/files.py [hunk 13]
```diff
@@ -0,0 +1,526 @@
...
+            'Data from the FABIAN HRTF data base shortened to 128 samples '
+            '(http://dx.doi.org/10.14279/depositonce-5718.5)')
+
+    if sampling_rate != 44100:
+        hrirs = pf.dsp.resample(hrirs, sampling_rate, 'freq', post_filter=True)
+
+    return hrirs, sources
+
+
+def rir(sampling_rate=48000):
```
Comment (mberz) [inline pyfar/signals/files.py:431]:
```suggestion
def room_impulse_response(sampling_rate=48000):
```

**Review by mberz — CHANGES_REQUESTED** [review]
Looks good to me.
I'd vote +10 for not using abbreviated function names though. ;)

**Comment by mberz** [comment]
Btw. Let's get rid of the commit history here by merging and squashing instead of simply merging. This should remove the binary files from history.

File: pyfar/signals/files.py (hunk 10 above)
Comment (f-brinkmann) [reply pyfar/signals/files.py]: done

File: pyfar/signals/files.py (hunk 11 above)
Comment (f-brinkmann) [reply pyfar/signals/files.py]: done

File: pyfar/signals/files.py (hunk 12 above)
Comment (f-brinkmann) [reply pyfar/signals/files.py]: done

File: pyfar/signals/files.py (hunk 13 above)
Comment (f-brinkmann) [reply pyfar/signals/files.py]: done

**Comment by f-brinkmann** [comment]
> Btw. Let's get rid of the commit history here by merging and squashing instead of simply merging. This should remove the binary files from history.

It's not the full dataset. Its a subset containing the horizontal and median plane. The file is 700 KB, which is comparable to the file size of the guitar sample.

**Comment by mberz** [comment]
Alright, but let's squash anyways.

**Comment by mberz** [comment]
@f-brinkmann 
pyfar/signals/files.py:529:80: E501 line too long (81 > 79 characters)

**Review by mberz — APPROVED** [review]

**Pull request merged**